token/metadata.go

```diff
@@ -61,17 +69,24 @@ func (m *Metadata) SpentTokenID() []*token.ID {
 // - The senders are included if and only if there is at least one output whose owner has the given enrollment IDs.
 // Application metadata is always included
 func (m *Metadata) FilterBy(eIDs ...string) (*Metadata, error) {
+	if m.TMS == nil || m.TokenRequestMetadata == nil {
+		return nil, errors.New("can't filter metadata by eID: nil TMS or nil token request metadata")
+	}
 	res := &Metadata{
 		TMS:                  m.TMS,
 		TokenRequestMetadata: &driver.TokenRequestMetadata{},
 	}
 
 	// filter issues
-	for _, issue := range m.TokenRequestMetadata.Issues {
+	for j, issue := range m.TokenRequestMetadata.Issues {
 		issueRes := driver.IssueMetadata{
 			Issuer: issue.Issuer,
 		}
 
+		if len(issue.ReceiversAuditInfos) != len(issue.Outputs) || len(issue.ReceiversAuditInfos) != len(issue.TokenInfo) ||
```

Comment:
Do we need this check here? Should we have a function that validates the metadata instead?

Reply:
This will be removed from here. A new issue will be opened to check well-formedness of token requests.